File: shared/src/commonMain/kotlin/net/brightroom/uniso/domain/identity/IdentityManager.kt
package net.brightroom.uniso.domain.identity

import net.brightroom.uniso.data.model.LocalUser
import net.brightroom.uniso.data.repository.SqlSettingsRepository
import net.brightroom.uniso.domain.account.currentTimestamp
import kotlin.uuid.ExperimentalUuidApi
import kotlin.uuid.Uuid

class IdentityManager(
    private val settingsRepository: SqlSettingsRepository,
) {
    @OptIn(ExperimentalUuidApi::class)
    fun getOrCreateLocalUserId(): String {
        val existing = settingsRepository.getLocalUser()
        if (existing != null) return existing.id

        val newId = Uuid.random().toString()
        settingsRepository.insertLocalUser(
            LocalUser(
                id = newId,
                createdAt = currentTimestamp(),
            ),
        )
        return newId
    }
}

File: packages/shared/src/domain/identity-manager.ts
import { SettingsRepository } from '../data/settings-repository'

export class IdentityManager {
  constructor(private settingsRepo: SettingsRepository) {}

  getOrCreateLocalUserId(): string {
    const existing = this.settingsRepo.getLocalUser()
    if (existing) return existing.id

    const id = crypto.randomUUID()
    const createdAt = new Date().toISOString()
    this.settingsRepo.insertLocalUser(id, createdAt)
    return id
  }
}
